### PR TITLE
chore: Removes toMap and associated functions

### PR DESCRIPTION
**Pull request by lantoli**
## Description

Removes toMap and associated functions.

It was tested successfully with Terraform repo: https://github.com/mongodb/terraform-provider-mongodbatlas/actions/runs/11241711780

Link to any related issue(s): CLOUDP-277767

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Required Checklist:

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added any necessary documentation (if appropriate)
- [x] I have run `make fmt` and formatted my code

### CODE REVIEW

File: admin/api_access_tracking.go [hunk 1]
```diff
@@ -170,8 +170,8 @@ func (a *AccessTrackingApiService) ListAccessLogsByClusterNameExecute(r ListAcce
 	}
 
 	localVarPath := localBasePath + "/api/atlas/v2/groups/{groupId}/dbAccessHistory/clusters/{clusterName}"
-	localVarPath = strings.Replace(localVarPath, "{"+"groupId"+"}", url.PathEscape(parameterValueToString(r.groupId, "groupId")), -1)
-	localVarPath = strings.Replace(localVarPath, "{"+"clusterName"+"}", url.PathEscape(parameterValueToString(r.clusterName, "clusterName")), -1)
+	localVarPath = strings.Replace(localVarPath, "{"+"groupId"+"}", url.PathEscape(r.groupId), -1)
```
Comment (lantoli) [inline admin/api_access_tracking.go:173]: we're always passing a string not a struct, no need to use parameterValueToString

File: tools/config/go-templates/client.mustache [hunk 2]
```diff
@@ -141,14 +126,6 @@ func parameterAddToHeaderOrQuery(headerOrQueryParams any, keyPrefix string, obj
 				value = "invalid"
 
 			case reflect.Struct:
-				if t,ok := obj.(modelWithMap); ok {
```
Comment (lantoli) [inline tools/config/go-templates/client.mustache:144]: toMap is not being implemented

Comment (wtrocki) [reply tools/config/go-templates/client.mustache:144]: None of the models implement it so this code does not execute. 
If we had any objects as parameters (currently that is not possible to express in swagger core) we would need to add JSON marshaler here. 

Separately. We initially had approach that minimized changes in the templates and flagged specific areas we change: https://github.com/mongodb/atlas-sdk-go/blob/main/tools/CONTRIBUTING.md#modification-of-the-custom-templates

This approach was to be able to continue update SDK with upstream. 
If your preference is to finetune templates we should likely remove {{! X-XGEN-MODIFIED }} and look to remove any dead code from templates (there is much more that we left in order to not make any unnecessary changes in templates.

File: tools/config/go-templates/api.mustache [hunk 3]
```diff
@@ -154,7 +154,7 @@ func (a *{{{classname}}}Service) {{{nickname}}}Execute(r {{operationId}}ApiReque
 	}
 
 	localVarPath := localBasePath + "{{{path}}}"{{#pathParams}}
-	localVarPath = strings.Replace(localVarPath, "{"+"{{baseName}}"+"}", url.PathEscape(parameterValueToString(r.{{paramName}}, "{{paramName}}")), -1){{/pathParams}}
+	localVarPath = strings.Replace(localVarPath, "{"+"{{baseName}}"+"}", url.PathEscape(r.{{paramName}}), -1){{/pathParams}}
```
Comment (wtrocki) [inline tools/config/go-templates/api.mustache:157]: I assume that you have tested all the type parameters we have in the API

**Review by wtrocki — APPROVED** [review]
LGTM. Haven't done testing

File: tools/config/go-templates/api.mustache (hunk 3 above)
Comment (lantoli) [reply tools/config/go-templates/api.mustache:157]: all TF tests passed and they have a good coverage.
also SDK wouldn't compile if some param were not a string, as it's required by strings.Replace

File: tools/config/go-templates/client.mustache (hunk 2 above)
Comment (wtrocki) [reply tools/config/go-templates/client.mustache:144]: SGTM

File: tools/config/go-templates/api.mustache (hunk 3 above)
Comment (wtrocki) [reply tools/config/go-templates/api.mustache:157]: That is odd. We do have integer and time params in API. It is just that in this structure they always represented as strings?  We definitely do have time based params. Very clever to use TF tests here

Comment (lantoli) [reply tools/config/go-templates/api.mustache:157]: note that this function is called only in path params, not in the body

**Comment by wtrocki** [comment]
As discussed. I think we can merge this PR and have follow up jira to rethink the approach for templates. 
Changes will improve code to not check for Object parameters as our API does not have them.

File: tools/config/go-templates/api.mustache (hunk 3 above)
Comment (wtrocki) [reply tools/config/go-templates/api.mustache:157]: Yes. Upstream SDK supports **Object** based parameters but they are rare in openapi and do not exist in Atlas.

**Pull request merged**